docs/source/tutorial/basic.rst

```diff
@@ -2,21 +2,586 @@
+
+.. code-block:: ipython
+
+    In [4]: from aiida.engine import run
+       ...: run(builder)
+
+Wait for the process to complete.
+Once it is done, it will return a dictionary with the output nodes.
+Exit the IPython shell and once more check for *all* processes:
+
+.. code-block:: console
+
+    $ verdi process list -a
+
+You should now see two processes in the list.
+One is the ``multiply`` calcfunction you ran earlier, the second is the ``ArithmeticAddCalculation`` CalcJob that you have just run.
+Grab the PK of the ``ArithmeticAddCalculation``, and generate the provenance graph.
+The result should look like the graph shown in :numref:`fig_calcjob_graph`.
+
+.. code-block:: console
+
+    $ verdi node graph generate 7
+
+.. _fig_calcjob_graph:
+.. figure:: include/calcjob_graph.png
+    :scale: 35
+    :align: center
+
+    Provenance graph of the ``ArithmeticAddCalculation`` CalcJob, with one input provided by the output of the ``multiply`` calculation function.
+
+You can see more details on any process, including its inputs and outputs, using the verdi shell:
+
+.. code:: console
+
+    $ verdi process show 7
+
+Submitting to the daemon
+========================
+
+When we used the ``run`` command in the previous section, the IPython shell was blocked while it was waiting for the ``CalcJob`` to finish.
+This is not a problem when we're simply adding two number together, but if we want to run multiple calculations that take hours or days, this is no longer practical.
+Instead, we are going to *submit* the ``CalcJob`` to the AiiDA *daemon*.
+The daemon is a program that runs in the background and manages submitted calculations until they are *terminated*.
+Let's first check the status of the daemon using the ``verdi`` CLI:
+
+.. code-block:: console
+
+    $ verdi daemon status
+
+If the daemon is running, the output will be something like the following:
+
+.. code-block:: bash
+
+    Profile: tutorial
+    Daemon is running as PID 96447 since 2020-05-22 18:04:39
+    Active workers [1]:
+      PID    MEM %    CPU %  started
+    -----  -------  -------  -------------------
+    96448    0.507        0  2020-05-22 18:04:39
+    Use verdi daemon [incr | decr] [num] to increase / decrease the amount of workers
+
+In this case, let's stop it for now:
+
+.. code-block:: console
+
+    $ verdi daemon stop
+
+Next, let's *submit* the ``CalcJob`` we ran previously.
+Start the ``verdi shell`` and execute the Python code snippet below.
+This follows all the steps we did previously, but now uses the ``submit`` function instead of ``run``:
+
+.. code-block:: ipython
+
+    In [1]: from aiida.engine import submit
+       ...:
+       ...: code = load_code(label='add')
+       ...: builder = code.get_builder()
+       ...: builder.x = load_node(pk=4)
+       ...: builder.y = Int(5)
+       ...:
+       ...: submit(builder)
+
+Note that the submission finishes very quickly, and that it returns the ``CalcJob`` that was just submitted:
```

Comment:
Why should we note that the submission finishes very quickly?

Reply:
I've rephrased this using one of the sentences that @sphuber suggested for the explanation on the difference between the ``run`` and ``submit`` functions for the how-to on running workflows.